### PR TITLE
Add react-toolbox sliders

### PR DESCRIPTION
**Pull request by bcldvd**
Would need a better way to handle change of CSS Vars

### CODE REVIEW

File: src/clients/mobiles/remote/playback/playback.tsx [hunk 1]
```diff
@@ -35,25 +39,32 @@ export class Playback extends React.Component<PlaybackPops, State>  {
                 timeAsPercentage: props.currentTimeAsPercentage
             })
         }
+        if (this.state.dominantBackgroundColor! = props.dominantBackgroundColor) {
+            this.setState({
+                dominantBackgroundColor: props.dominantBackgroundColor
+            })
+
+            changeCSSVar('progress-main-color', this.state.dominantBackgroundColor);
```
Comment (wdelmas) [inline src/clients/mobiles/remote/playback/playback.tsx]: Not the good way to update style ;)

File: src/clients/mobiles/remote/playback/playback.tsx [hunk 2]
```diff
@@ -35,25 +39,32 @@ export class Playback extends React.Component<PlaybackPops, State>  {
...
         if (this.props.onTimeChange)
             this.props.onTimeChange(time)
     }
+
     public render() {
         return (
             <div className={styles.timeSlider}>
-                <input ref="time" type="range" min="1" max="100"
-                    value={this.state.timeAsPercentage.toString()}
+                <Slider value={this.state.timeAsPercentage}
```
Comment (wdelmas) [inline src/clients/mobiles/remote/playback/playback.tsx]: You can get rid of the manually binding if you convert  ontimechanged to pure puction ( not an array function )

File: src/clients/mobiles/remote/volume/volume.tsx [hunk 3]
```diff
@@ -38,14 +39,14 @@ export class Volume extends React.Component<VolumePops, State>  {
         }
     }
 
-    public _onVolumeChanged = () => {
-        const value = parseInt(this.refs.volume.value) / 100
+    public _onVolumeChanged = (value: number) => {
+        // debugger;
```
Comment (wdelmas) [inline src/clients/mobiles/remote/volume/volume.tsx]: Comment

File: src/clients/mobiles/remote/playback/playback.tsx [hunk 4]
```diff
@@ -35,25 +39,32 @@ export class Playback extends React.Component<PlaybackPops, State>  {
...
+            changeCSSVar('progress-main-color', this.state.dominantBackgroundColor);
+            changeCSSVar('slider-main-color', this.state.dominantBackgroundColor);
+        }
     }
 
-    public _onTimeChange = () => {
-        const timeAsPercentage = parseInt(this.refs.time.value)
-        const time = timeAsPercentage* this.props.duration /100
+    public _onTimeChange = (value: number) => {
+        const time = (value * this.props.duration) / 100
```
Comment (wdelmas) [inline src/clients/mobiles/remote/playback/playback.tsx]: Check if value defined

File: src/clients/mobiles/remote/playback/playback.tsx (hunk 2 above)
Comment (bcldvd) [reply src/clients/mobiles/remote/playback/playback.tsx]: Actually ```() => this._onTimeChange()``` doesn't work with the way sliders are handled in [React Toolbox](http://react-toolbox.com/#/components/slider)
If you see how to enable it let me know !

File: src/clients/mobiles/remote/playback/playback.tsx [hunk 5]
```diff
@@ -50,6 +50,7 @@ export class Playback extends React.Component<PlaybackPops, State>  {
     }
 
     public _onTimeChange = (value: number) => {
+        if (typeof value == 'undefined') return;
```
Comment (wdelmas) [inline src/clients/mobiles/remote/playback/playback.tsx]: if(!value) should be enough :)

**Comment by wdelmas** [comment]
I'll really don't like to work with such big frontend library....
From what I've seen, you need to wrap you root component with `<ThemeProvider theme={theme}>` .
IMO, we should look at another library. (https://github.com/react-component/slider)

**Review by wdelmas — COMMENTED** [review]
see my last comment

File: src/clients/mobiles/remote/playback/playback.tsx (hunk 5 above)
Comment (bcldvd) [reply src/clients/mobiles/remote/playback/playback.tsx]: Not if the value is 0 🤔

**Comment by bcldvd** [comment]
I'm up for it, but I thought we'd be using other components in the end ;)
Your call, I'm not familiar with React libraries, though Material UI seemed used everywhere and might be a good solution

File: src/clients/mobiles/remote/playback/playback.tsx (hunk 5 above)
Comment (wdelmas) [reply src/clients/mobiles/remote/playback/playback.tsx]: I'm not sure but you'll get "0"

**Pull request merged**